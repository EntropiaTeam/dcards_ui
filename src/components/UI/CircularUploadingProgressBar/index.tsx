import React, { FC } from 'react';
import clsx from 'clsx';
import { makeStyles } from '@material-ui/core/styles';
import { Box, CircularProgress, Typography } from '@material-ui/core';
import TranslatedTypography from '../TranslatedTypography';

const useStyles = makeStyles({
  general: {
    position: 'absolute',
    top: 0,
    right: 0,
    left: 0,
    bottom: 0,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  root: {
    zIndex: 1000
  },
  container: {
    position: 'relative',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%'
  },
  progress: {
    color: 'white'
  },
  translate: {
    top: 140,
    color: 'white'
  }
});

const CircularUploadingProgressBar: FC<{ progress: number }> = ({ progress }) => {
  const classes = useStyles();

  const isUploading = progress !== 100;
  const roundedProgress = `${Math.round(progress)}%`;

  const styles = (className: string) => clsx(classes.general, className);

  const uploadingProgress = (
    <>
      <CircularProgress size={100} value={progress} color="primary" variant="determinate" />
      <Typography variant="body1" color="inherit" className={styles(classes.progress)}>
        {roundedProgress}
      </Typography>
      <TranslatedTypography
        i18nKey="processes.upload"
        classes={{ root: styles(classes.translate) }}
      />
    </>
  );

  const processingSpinner = (
    <>
      <CircularProgress size={100} value={progress} color="primary" />
      <TranslatedTypography
        i18nKey="processes.processing"
        classes={{ root: styles(classes.translate) }}
      />
    </>
  );

  return (
    <Box className={styles(classes.root)}>
      <Box className={classes.container}>
        {isUploading ? uploadingProgress : processingSpinner}
      </Box>
    </Box>
  );
};

export default CircularUploadingProgressBar;
