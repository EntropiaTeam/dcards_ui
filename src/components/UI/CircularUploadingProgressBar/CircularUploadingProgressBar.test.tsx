import React from 'react';
import renderer from 'react-test-renderer';
import CircularUploadingProgressBar from '.';

interface IntlMock {
  useTranslation: () => void;
}

interface TranslationMock {
  t: () => void;
}

jest.mock('react-i18next', (): IntlMock => ({
  useTranslation: (): TranslationMock => ({
    t: (key: string): string => key
  })
}));

describe('<CircularUploadingProgressBar />', () => {
  const CircularUploadingProgressBarComponent = renderer.create(
    <CircularUploadingProgressBar />
  ).toJSON();

  it('should render CircularUploadingProgressBar component correctly', () => {
    expect(CircularUploadingProgressBarComponent).toMatchSnapshot();
  });
});
