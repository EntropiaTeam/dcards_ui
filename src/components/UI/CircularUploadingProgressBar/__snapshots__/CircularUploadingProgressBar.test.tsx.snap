// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircularUploadingProgressBar /> should render CircularUploadingProgressBar component correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-6 makeStyles-general-1 makeStyles-root-2"
>
  <div
    className="MuiBox-root MuiBox-root-7 makeStyles-container-3"
  >
    <div
      aria-valuenow={0}
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
      role="progressbar"
      style={
        Object {
          "height": 100,
          "transform": "rotate(-90deg)",
          "width": 100,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={
            Object {
              "strokeDasharray": "126.920",
              "strokeDashoffset": "126.920px",
            }
          }
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root makeStyles-general-1 makeStyles-progress-4 MuiTypography-body1 MuiTypography-colorInherit"
    >
      NaN%
    </p>
    <p
      className="MuiTypography-root makeStyles-general-1 makeStyles-translate-5 MuiTypography-body1"
    >
      processes.upload
    </p>
  </div>
</div>
`;
